enhancements/kubelet/user-namespaces-support.md

```diff
@@ -0,0 +1,330 @@
+the same as the one inside the container.
+
+Note: upstream Kubernetes deemed `runAsUser` to be a safe field to use for pods with `hostUsers: false`. However, the philosophy
+of SCC utilizes defense-in-depth. `restricted-v3` will be the most restrictive policy.
+
+##### nonroot-v3
+
+This SCC profile will be identical to `nonroot-v2`, except it will set `HostUser` to `Disallowed`, thus forcing pods to be in a
+user namespace. It differs from `restricted-v3` because it allows the fields `runAsUser`/`runAsGroup`/`fsGroup`.
+
+Note: a user can actually request `runAsUser: 0` in this profile, despite its name being `nonroot`. This is because root inside
+the container is not root on the host, and because `anyuid` SCC allows privilege escalation, which we deemed out of scope for this SCC.
+
+##### nested-container
+
+The intention of this SCC is to allow a user to run `podman` or other container engine inside of an Openshift pod.
+Since user namespaces allow a process to gain access to the capabilities needed in a safe way, it's a natural addition
+to the proposal of adding user namespaces generally.
+
+This SCC will largely mirror the `restricted-v2` SCC, but have a couple of changes.
+- SELinux context set to MustRunAs.SeLinuxOptions.Type: `container_engine_t`
+    - This SELinux type has been developed to allow the majority of podman in pod situations, and can
+      continue to be adapted without affecting the normal `container_t` which should be more restrictive.
+- RunAsUserStrategy: RunAsAny
+    - Any user should be allowed, as the user running in the container is not the same running outside.
+- RequiredDropCapabilities: None
+    - Inside of a user namespace, the capabilities a pod requests are only present in the user namespace,
+      not on the host. Thus, even for a less trusted user, the capabilities should be safe to access.
+- HostUser set to Disallowed
+
+Note: to use this SCC, the namespace must be labeled as `privileged` in Pod Security Standards. This is in part because
+PSS doesn't recognized `container_engine_t` [in 1.30](https://github.com/kubernetes/kubernetes/pull/126165), and in part because
+the baseline policy doesn't allow Unmasked procMount, even if the pod is in a user namespace, [in 1.30](https://github.com/kubernetes/kubernetes/pull/126163).
+
+These are fixed in Kubernetes 1.31, and Openshift 4.18, meaning a namespace can be labeled as `baseline` and successfully utilize all the fields of the `nested-container` SCC.
+
+#### Feature Gates and Sets
+
+There are three feature gates of note for this enhancement: `UserNamespacesSupport`, `UserNamespacesPodSecurityStandards`, and `ProcMountType`.
+
+`UserNamespacesSupport` will be on by default OOTB in 4.17, as it's beta in 1.30. It is the main featuregate of this feature.
+Enabling `UserNamespacesPodSecurityStandards` and `ProcMountType` will move a cluster into TechPreviewNoUpgrade in 4.17.
```

Comment:
What is `ProcMountType`?

Reply:
I added a section in the feature gates section that describes each feature gate